service/controller/v24/adapter/cpf_route_tables.go

```diff
@@ -0,0 +1,105 @@
+package adapter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/v24/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/v24/encrypter"
+	"github.com/giantswarm/aws-operator/service/controller/v24/key"
+)
+
+type CPFRouteTables struct {
+	PrivateRoutes []CPFRouteTablesRoute
+	PublicRoutes  []CPFRouteTablesRoute
+}
+
+func (a *CPFRouteTables) Adapt(ctx context.Context, config Config) error {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var tenantPrivateSubnetCidrs []string
+	{
+		for _, az := range key.StatusAvailabilityZones(config.CustomObject) {
+			tenantPrivateSubnetCidrs = append(tenantPrivateSubnetCidrs, az.Subnet.Private.CIDR)
+		}
+	}
+
+	// private routes.
+	for _, routeTableName := range config.CustomObject.Spec.AWS.VPC.RouteTableNames {
+		routeTableID, err := routeTableID(routeTableName, config)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		for _, cidrBlock := range tenantPrivateSubnetCidrs {
+			rt := CPFRouteTablesRoute{
+				RouteTableName: routeTableName,
+				RouteTableID:   routeTableID,
+				// Requester CIDR block, we create the peering connection from the guest's private subnets.
+				CidrBlock:        cidrBlock,
+				PeerConnectionID: cc.Status.Cluster.VPCPeeringConnectionID,
```

Comment:
We take the ID from the controller context now. The value is fetched from the stack outputs in the `stackoutput` resource.

Reply:
Adding comment about `The value is fetched from the stack outputs in the stackoutput resource.` could be useful here.

Reply:
I added the comments.